types/vote.go

```diff
@@ -202,14 +203,21 @@ func (vote *Vote) String() string {
 		panic("Unknown vote type")
 	}
 
-	return fmt.Sprintf("Vote{%v:%X %v/%02d/%v(%v) %X %X %X @ %s}",
+	var blockHashString string
+	if len(vote.BlockID.Hash) > 0 {
+		blockHashString = fmt.Sprintf("%X", tmbytes.Fingerprint(vote.BlockID.Hash))
+	} else {
+		blockHashString = nilVoteStr
+	}
+
+	return fmt.Sprintf("Vote{%v:%X %v/%02d/%v(%v) %s %X %X @ %s}",
 		vote.ValidatorIndex,
 		tmbytes.Fingerprint(vote.ValidatorAddress),
 		vote.Height,
 		vote.Round,
 		vote.Type,
 		typeString,
```

Comment:
Addressed, I used a slightly different format thought since it felt more readable:
```
Vote{%v:%X %v/%02d %s %s %X %X @ %s}
```

Please let me know if that's okay

Reply:
yeah that's fine. I'm still not sure why we have `%02d` instead of just `%d` - won't this potentially truncate the round this is in?

Reply:
@cmwaters It just pads values less than 10 to be of the form `07`. Other values will display correctly. I'm not really sure this is necessary.

https://go.dev/play/p/fKRbm_Qs8UN

Reply:
Changed to just `%d`